langchain/src/document_loaders/pdf.ts

```diff
@@ -0,0 +1,34 @@
+import type { readFile as ReadFileT } from "fs/promises";
+// the main entrypoint has some debug code that we don't want to import
+import pdf from "pdf-parse/lib/pdf-parse.js";
+import { Document } from "../document.js";
+import { getEnv } from "../util/env.js";
+import { BaseDocumentLoader } from "./base.js";
+
+export class PDFLoader extends BaseDocumentLoader {
+  constructor(public filePath: string) {
+    super();
+  }
+
+  public async load(): Promise<Document[]> {
+    const { readFile } = await PDFLoader.imports();
+    const buffer = await readFile(this.filePath);
+    const parsed = await pdf(buffer);
+    const metadata = { source: this.filePath };
+    return [new Document({ pageContent: parsed.text, metadata })];
+  }
```

Comment:
would it be good to provide a way to load from a buffer directly? I think there might be use cases such as upload a pdf file from a form from frontend and it will not be from file system. https://github.com/fraserxu/book-gpt/blob/main/pages/api/ingest.ts#L43

Reply:
Very much agree, and for every file-based loader, in a future PR touching all of them

Reply:
@nfcampos any updates / plans on this?